pkg/authenticator/authenticator.go

```diff
@@ -200,16 +200,44 @@ func (auth *Authenticator) IsCertExpired() bool {
 	return currentDate.Add(bufferTime).After(certExpiresOn)
 }
 
-// Authenticate sends Conjur an authenticate request and returns
-// the response data. Also manages state of certificates.
-func (auth *Authenticator) Authenticate() ([]byte, error) {
+// Authenticate sends Conjur an authenticate request and writes the response
+// to the token file (after decrypting it if needed). It also manages state of
+// certificates.
+func (auth *Authenticator) Authenticate() error {
 	log.Info(log.CAKC040, auth.Config.Username)
 
+	err := auth.loginIfNeeded()
+	if err != nil {
+		return err
+	}
+
+	authenticationResponse, err := auth.makeAuthenticationRequest()
+	if err != nil {
+		return err
+	}
+
+	parsedResponse, err := auth.parseAuthenticationResponse(authenticationResponse)
+	if err != nil {
+		return log.RecordedError(log.CAKC020)
+	}
+
+	err = auth.AccessToken.Write(parsedResponse)
+	if err != nil {
+		return err
+	}
+
+	log.Info(log.CAKC035)
+	return nil
+}
+
+// loginIfNeeded checks if we need to send a login request to Conjur and sends
+// one if needed
+func (auth *Authenticator) loginIfNeeded() error {
 	if !auth.IsLoggedIn() {
 		log.Debug(log.CAKC039)
 
 		if err := auth.Login(); err != nil {
-			return nil, log.RecordedError(log.CAKC015)
+			return log.RecordedError(log.CAKC015)
```

Comment:
Not a blocker but it's a bit strange that we don't use the returned errors for neither `CAKC015` nor `CAKC020` in any way

Reply:
wdym? we log the message here and also raise it in the `Authenticate()` method